engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/CompilationResult.scala

```diff
@@ -45,4 +58,17 @@ object CompilationResult extends Applicative[CompilationResult] {
       }
     }
   }
-}
+
+  implicit def takingLastNodeTypingInfoSemigroup: Semigroup[ValidationContext] = new Semigroup[ValidationContext] with LazyLogging {
+    override def combine(x: ValidationContext, y: ValidationContext): ValidationContext = {
+      logger.whenWarnEnabled {
+        if (x != y) {
+          logger.warn(s"Merging different ValidationContext for the same nodes: $x != $y. This can be a bug in code or duplicated node ids with different node typing info")
+        }
+      }
+      // we should be lax here because we won't to detect duplicate nodes and context can be different than
```

Comment:
I think some words are missing from this comment :)

Reply:
It should be "then" instead "than". Have it sense?